diff --git a/glib/glibmm.pc.in b/glib/glibmm.pc.in
index b08f62d..25e6c16 100644
--- a/glib/glibmm.pc.in
+++ b/glib/glibmm.pc.in
@@ -16,6 +16,6 @@ Name: glibmm
 Description: C++ wrapper for GLib
 Version: @PACKAGE_VERSION@
 URL: http://www.gtkmm.org/
-Requires: gobject-2.0 sigc++-2.0
+Requires: gobject-2.0 sigc++-3.0
 Libs: -L${libdir} -lglibmm-@GLIBMM_API_VERSION@
 Cflags: -I${includedir}/@GLIBMM_MODULE_NAME@ -I${libdir}/@GLIBMM_MODULE_NAME@/include
