diff --git a/re2/re2.cc b/re2/re2.cc
index 61d9d1f..284ae39 100644
--- a/re2/re2.cc
+++ b/re2/re2.cc
@@ -23,6 +23,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <iostream>
 
 #include "absl/base/macros.h"
 #include "absl/container/fixed_array.h"
@@ -36,6 +37,47 @@
 #include "re2/sparse_array.h"
 
 namespace re2 {
+StringWrapper::StringWrapper() : inner_() {}
+StringWrapper::StringWrapper(absl::string_view s) : inner_(s) {}
+StringWrapper::StringWrapper(std::string &&s) : inner_(s) {}
+
+StringWrapper &StringWrapper::operator=(absl::string_view s) {
+  inner_ = std::string(s);
+  return *this;
+}
+StringWrapper &StringWrapper::operator=(std::string &&s) {
+  inner_ = s;
+  return *this;
+}
+
+StringWrapper::~StringWrapper() {
+  std::cerr << "destroyed string wrapper with len=" << size() << std::endl;
+}
+
+bool StringWrapper::empty() const { return inner_.empty(); }
+
+size_t StringWrapper::size() const { return inner_.size(); }
+
+absl::string_view StringWrapper::view() const { return inner_; }
+
+std::string* StringWrapper::get_mutable() { return &inner_; }
+
+GroupNames::GroupNames(const RE2 &r) : names_() {
+  for (const auto &n : r.NamedCapturingGroups()) {
+    assert(n.second > 0);
+    const size_t n2 = n.second;
+    names_.emplace_back(n.first, n2);
+  }
+}
+GroupNames::~GroupNames() {
+  std::cerr << "destroyed group names with len=" << size() << std::endl;
+}
+
+size_t GroupNames::size() const { return names_.size(); }
+
+const NamedGroup& GroupNames::at(size_t i) const {
+  return names_.at(i);
+}
 
 // Controls the maximum count permitted by GlobalReplace(); -1 is unlimited.
 static int maximum_global_replace_count = -1;
@@ -197,11 +239,15 @@ int RE2::Options::ParseFlags() const {
   return flags;
 }
 
-void RE2::Init(absl::string_view pattern, const Options& options) {
+absl::string_view RE2::pattern_view() const { return *pattern_; }
+
+absl::string_view RE2::error_view() const { return *error_; }
+
+absl::string_view RE2::error_arg_view() const { return *error_arg_; }
+
+void RE2::Init(absl::string_view pattern, const Options &options) {
   static absl::once_flag empty_once;
-  absl::call_once(empty_once, []() {
-    (void) new (empty_storage) EmptyStorage;
-  });
+  absl::call_once(empty_once, []() { (void)new (empty_storage) EmptyStorage; });
 
   pattern_ = new std::string(pattern);
   options_.Copy(options);
diff --git a/re2/re2.h b/re2/re2.h
index 68fbed1..0126044 100644
--- a/re2/re2.h
+++ b/re2/re2.h
@@ -227,7 +227,48 @@
 namespace re2 {
 class Prog;
 class Regexp;
-}  // namespace re2
+class RE2;
+
+class StringWrapper {
+public:
+  StringWrapper();
+  StringWrapper(absl::string_view s);
+  StringWrapper(std::string&& s);
+
+  StringWrapper &operator=(absl::string_view s);
+  StringWrapper &operator=(std::string&& s);
+
+  ~StringWrapper();
+
+  bool empty() const;
+  size_t size() const;
+  absl::string_view view() const;
+  std::string* get_mutable();
+
+private:
+  std::string inner_;
+};
+
+struct NamedGroup {
+  absl::string_view name;
+  size_t index;
+
+  NamedGroup(absl::string_view s, size_t i): name(s), index(i) {}
+  NamedGroup(NamedGroup&& rhs): name(std::move(rhs.name)), index(std::move(rhs.index)) {}
+  NamedGroup& operator=(const NamedGroup& rhs) = default;
+};
+
+class GroupNames {
+public:
+  GroupNames(const RE2& r);
+  ~GroupNames();
+  size_t size() const;
+  const NamedGroup& at(size_t i) const;
+
+private:
+  std::vector<NamedGroup> names_;
+};
+} // namespace re2
 
 namespace re2 {
 
@@ -309,10 +350,14 @@ class RE2 {
   //   re.pattern();    // "ab*c?d+"
   const std::string& pattern() const { return *pattern_; }
 
+  absl::string_view pattern_view() const;
+
   // If RE2 could not be created properly, returns an error string.
   // Else returns the empty string.
   const std::string& error() const { return *error_; }
 
+  absl::string_view error_view() const;
+
   // If RE2 could not be created properly, returns an error code.
   // Else returns RE2::NoError (== 0).
   ErrorCode error_code() const { return error_code_; }
@@ -321,6 +366,8 @@ class RE2 {
   // portion of the regexp.
   const std::string& error_arg() const { return *error_arg_; }
 
+  absl::string_view error_arg_view() const;
+
   // Returns the program size, a very approximate measure of a regexp's "cost".
   // Larger numbers are more expensive than smaller numbers.
   int ProgramSize() const;
diff --git a/re2/testing/re2_test.cc b/re2/testing/re2_test.cc
index 151525f..1fd42ce 100644
--- a/re2/testing/re2_test.cc
+++ b/re2/testing/re2_test.cc
@@ -1658,4 +1658,76 @@ TEST(RE2, Issue310) {
   ASSERT_EQ(m, "") << " got m='" << m << "', want ''";
 }
 
-}  // namespace re2
+TEST(RE2, StringWrapper) {
+  // We can use StringWrapper to wrap std::string.
+
+  StringWrapper s;
+
+  ASSERT_EQ(s.view(), "");
+  ASSERT_TRUE(s.empty());
+
+  s = absl::string_view("asdf");
+  ASSERT_EQ(s.view(), "asdf");
+  ASSERT_FALSE(s.empty());
+
+  s = std::string("bbbb");
+  ASSERT_EQ(s.view(), "bbbb");
+  ASSERT_FALSE(s.empty());
+
+  *s.get_mutable() = "cccc";
+  ASSERT_EQ(s.view(), "cccc");
+  ASSERT_FALSE(s.empty());
+}
+
+TEST(RE2, GroupNames) {
+  // We can generate group names.
+
+  RE2 no_groups("asdf");
+  RE2 num_group("(a)sdf");
+  RE2 named_group("(?P<foo>a)sdf");
+  RE2 mixed_groups("(?P<foo>a)s(d)f");
+
+  ASSERT_TRUE(RE2::FullMatch("asdf", no_groups));
+  GroupNames g1(no_groups);
+  ASSERT_EQ(0, g1.size());
+
+  std::string s;
+
+  ASSERT_TRUE(RE2::FullMatch("asdf", num_group, &s));
+  ASSERT_EQ("a", s);
+  s.clear();
+  GroupNames g2(num_group);
+  ASSERT_EQ(0, g2.size());
+
+  ASSERT_TRUE(RE2::FullMatch("asdf", named_group, &s));
+  ASSERT_EQ("a", s);
+  s.clear();
+
+  const auto &m3 = named_group.NamedCapturingGroups();
+  ASSERT_EQ(1, m3.size());
+  ASSERT_EQ(1, m3.at("foo"));
+
+  GroupNames g3(named_group);
+  ASSERT_EQ(1, g3.size());
+  ASSERT_EQ("foo", g3.at(0).name);
+  ASSERT_EQ(1, g3.at(0).index);
+
+  std::string t;
+
+  ASSERT_TRUE(RE2::FullMatch("asdf", mixed_groups, &s, &t));
+  ASSERT_EQ("a", s);
+  s.clear();
+  ASSERT_EQ("d", t);
+  t.clear();
+
+  const auto &m4 = mixed_groups.NamedCapturingGroups();
+  ASSERT_EQ(1, m4.size());
+  ASSERT_EQ(1, m4.at("foo"));
+
+  GroupNames g4(named_group);
+  ASSERT_EQ(1, g4.size());
+  ASSERT_EQ("foo", g4.at(0).name);
+  ASSERT_EQ(1, g4.at(0).index);
+}
+
+} // namespace re2
